### PR TITLE
Allow configuration of maxDepthOfQuery

### PR DESCRIPTION
**Pull request by raymondfeng**
https://github.com/strongloop/loopback-datasource-juggler/issues/1651

#### Related issues

- connect to #1653 #1651

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: test/model-definition.test.js [hunk 1]
```diff
@@ -257,6 +257,35 @@ describe('ModelDefinition class', function() {
     done();
   });
 
+  it('should report errors for deep query than maxDepthOfQuery', function(done) {
+    var MyModel = memory.createModel('my-model', {}, {
+      maxDepthOfQuery: 5,
+    });
+
+    var filter = {where: {and: [{and: [{and: [{and: [{and: [{and:
```
Comment (b-admike) [inline test/model-definition.test.js]: nitpick: In the future, if we have more tests like these, we can extract this part out to a helper function like `givenFilterWithDepth()` or something like that for better readability.

File: lib/utils.js [hunk 2]
```diff
@@ -336,18 +336,25 @@ function sanitizeQuery(query, options) {
...
      * Security risk if the client passes in a very deep where object
      */
+    if (this.circular) {
+      const msg = g.f('The query object is circular');
+      const err = new Error(msg);
+      err.statusCode = 400;
+      err.code = 'QUERY_OBJECT_IS_CIRCULAR';
+      throw err;
+    }
     if (this.level > maxDepth || this.circular) {
```
Comment (b-admike) [inline lib/utils.js]: Can we remove the check for `this.circular` in this if condition here since we've separated the checks?

**Comment by b-admike** [comment]
@raymondfeng I think we should also include some docs about the usage of this new flag for users opting to customize it for their applications.

**Comment by raymondfeng** [comment]
Docs - https://github.com/strongloop/loopback.io/pull/760

File: lib/utils.js (hunk 2 above)
Comment (raymondfeng) [reply lib/utils.js]: fixed

**Comment by raymondfeng** [comment]
@b-admike PTAL

**Pull request merged**